documentation/library_guide/parallel_api/pass_data_algorithms.rst

```diff
@@ -81,47 +76,65 @@ the buffer were created for the same queue. For example:
+When using device USM, such as allocated by ``malloc_device``, manually copy data to this memory
+before calling oneDPL algorithms, and copy it back once the algorithms have finished execution.
+
+Use std::vector
+-----------------------------
+
+The following examples demonstrate two ways to use the parallel algorithms with ``std::vector``:
+
+* host allocators
+* USM allocators
+
+You can use iterators to host allocated ``std::vector`` data
+as shown in the following example:
 
 .. code:: cpp
 
   #include <oneapi/dpl/execution>
   #include <oneapi/dpl/algorithm>
-  #include <sycl/sycl.hpp>
+  #include <vector>
   int main(){
-    const int n = 1000;
-    auto policy = oneapi::dpl::execution::dpcpp_default;
-    sycl::usm_allocator<int, sycl::usm::alloc::shared> alloc(policy.queue());
-    std::vector<int, decltype(alloc)> vec(n, alloc);
-
-    std::fill(policy, vec.begin(), vec.end(), 42);
-
+    std::vector<int> vec( 1000 );
+    std::fill(oneapi::dpl::execution::dpcpp_default, vec.begin(), vec.end(), 42);
+    // each element of vec equals to 42
     return 0;
   }
 
-When using device USM, such as allocated by ``malloc_device``, manually copy data to this memory
-before calling oneDPL algorithms, and copy it back once the algorithms have finished execution.
-
-Use Host-Side std::vector
------------------------------
+When using iterators to host allocated data, a temporary SYCL buffer is created, and the data
+is copied to this buffer. After processing on a device is complete, the modified data is copied
+from the temporary buffer back to the host container. While convenient, using host allocated
+data can lead to unintended copying between host and device. We recommend working with SYCL buffers
+or USM memory to reduce data copying between the host and device. 
 
-|onedpl_long| parallel algorithms can be called with ordinary (host-side) iterators, as seen in the
-example below.
-In this case, a temporary SYCL buffer is created, and the data is copied to this buffer.
-After processing on a device is complete, the modified data is copied from the temporary buffer back
-to the host container.
-For example:
+You can also use ``std::vector`` with a USM allocator, as shown in the following example:
 
 .. code:: cpp
 
   #include <oneapi/dpl/execution>
   #include <oneapi/dpl/algorithm>
-  #include <vector>
+  #include <sycl/sycl.hpp>
   int main(){
-    std::vector<int> vec( 1000 );
-    std::fill(oneapi::dpl::execution::dpcpp_default, vec.begin(), vec.end(), 42);
-    // each element of vec equals to 42
+    const int n = 1000;
+    auto policy = oneapi::dpl::execution::dpcpp_default;
+    sycl::usm_allocator<int, sycl::usm::alloc::shared> alloc(policy.queue());
+    std::vector<int, decltype(alloc)> vec(n, alloc);
+
+    // Recommended to use USM pointers:
+    std::fill(policy, vec.data(), vec.data() + vec.size(), 42);
+
+    // Iterators for USM allocators might require extra copying
```

Comment:
1) I would add clarification  `not recommended way`:
`// Iterators for USM allocators might require extra copying - not recommended way`
2) BTW, the overloads of  `dpl::begin`, `dpl::end` for `std::vector` (see my proposal above) could always return pointer type and encapsulate memory type RT check and create a memory buffer if is necessary.

Reply:
fixed (1).